supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala

```diff
@@ -246,35 +381,93 @@ final case class AwsBatchJob(jobDescriptor: BackendJobDescriptor, // WDL/CWL
+    val describeComputeEnvironmentsResponse = batchClient.describeComputeEnvironments( DescribeComputeEnvironmentsRequest.builder().computeEnvironments(computeEnvironments.asJava).build())
+    val ecsClusterArns = describeComputeEnvironmentsResponse.computeEnvironments().asScala.map(_.ecsClusterArn())
+
+    val ecsClient = configureClient(EcsClient.builder(), optAwsAuthMode, configRegion)
+
+    val instanceIds: Seq[String] = ecsClusterArns.map(containerArn => ecsClient.describeContainerInstances(DescribeContainerInstancesRequest.builder().containerInstances(containerInstanceArn).cluster(containerArn).build()))
+      .map(r => r.containerInstances().asScala).flatMap(_.map(_.ec2InstanceId()))
+
+    BatchJobContainerContext(jobId, containerInstanceArn, ecsClusterArns, instanceIds)
+  }
+
+  case class BatchJobContainerContext(jobId: String, containerInstanceArn: String, ecsClusterArns: Seq[String], ec2InstanceIds: Seq[String]) {
+    override def toString: String = {
+      new ToStringBuilder(this, ToStringStyle.JSON_STYLE)
+        .append("jobId", this.jobId)
+        .append("containerInstanceArn", containerInstanceArn)
+        .append("ecsClusterArns", ecsClusterArns)
+        .append("ec2InstanceIds", ec2InstanceIds)
+        .build()
+    }
   }
 
-  //TODO: unused at present
   def rc(detail: JobDetail): Integer = {
      detail.container.exitCode
   }
 
+  //todo: unused at present??
   def output(detail: JobDetail): String = {
-     val events: Seq[OutputLogEvent] = logsclient.getLogEvents(GetLogEventsRequest.builder
+     val events: Seq[OutputLogEvent] = cloudWatchLogsClient.getLogEvents(GetLogEventsRequest.builder
                                             // http://aws-java-sdk-javadoc.s3-website-us-west-2.amazonaws.com/latest/software/amazon/awssdk/services/batch/model/ContainerDetail.html#logStreamName--
                                             .logGroupName("/aws/batch/job")
                                             .logStreamName(detail.container.logStreamName)
                                             .startFromHead(true)
-                                            .build).events.asScala.toSeq
+                                            .build).events.asScala
      val eventMessages = for ( event <- events ) yield event.message
      eventMessages mkString "\n"
   }
 
   //TODO: Wrap in cats Effect
   def abort(jobId: String): CancelJobResponse = {
-    client.cancelJob(CancelJobRequest.builder.jobId(jobId).reason("cromwell abort called").build)
+    batchClient.cancelJob(CancelJobRequest.builder.jobId(jobId).reason("cromwell abort called").build)
+  }
+
+  /**
+    * Generate a `String` describing the instance. Mainly for debugging
+    * @return a description of the instance
+    */
+  override def toString: String = {
+     new ToStringBuilder(this, ToStringStyle.JSON_STYLE)
```

Comment:
TIL about this layout style - very cool. It's something I've spent much time manually hacking about trying to emulate in the past so I'm glad to find out there's an easier way!

Reply:
ToStringBuilder and ToStringStyle are my friends